### ISSUE TITLE
Destination: Digitalocean spaces

### ISSUE THREAD

**Issue by JannikZed** [rossigee/backups#10]
Hi there,
I was trying to use your backup system. Unfortunately, I can't backup to a generic S3 destination like digitalocean spaces.
Is it planned to support other S3 destinations as well?

**Comment by rossigee** [comment]
Yes, support for non-Amazon S3-like destinations (i.e. Minio) is desirable, and something I plan to look into at some point.

What issue are you facing with the configuration of your DigitalOcean spaces buckets?

**Comment by JannikZed** [comment]
I think just the configuration option for the S3 host is missing right now. Otherwise, you can only connect to amazon servers.

**Comment by rossigee** [comment]
I've added an 'endpoint_url' parameter. Does that help?

https://github.com/rossigee/backups/commit/078f57543acb3a70b28dfc9a89fa95079fc079ca

**Comment by rossigee** [comment]
Implemented a native Minio destination driver in backups/destinations/minio.py using the `minio` Python SDK. This works with Minio, DigitalOcean Spaces, and any S3-compatible service that has a Minio-compatible endpoint. Configuration example:

```json
{
  "id": "minio-backup",
  "type": "minio",
  "endpoint": "nyc3.digitaloceanspaces.com",
  "bucket": "my-backup-bucket",
  "secure": true,
  "credentials": {
    "access_key": "your-access-key",
    "secret_key": "your-secret-key"
  },
  "retention_copies": 5
}
```
